File: python/richest-customer-wealth.py
from typing import List


class Solution:
    # Time complexity: O(n + m) where n and m are the number of customers and banks
    # Space complexity: O(1)
    def maximumWealth(self, accounts: List[List[int]]) -> int:
        res = 0
        for customer in accounts:
            curr = 0
            for bank in customer:
                curr += bank
            res = max(res, curr)
        return res
    
    # Time complexity: O(n + m) where n and m are the number of customers and banks
    # Space complexity: O(1)
    def maximumWealth2(self, accounts: List[List[int]]) -> int:
        return max(sum(customer) for customer in accounts)

File: go/richest_customer_wealth.go
package leetcode

// Time complexity: O(n * m) where n and m are the number of clients and banks
// Space complexity: O(1)
func maximumWealth(accounts [][]int) int {
	res := 0
	for _, client := range accounts {
		res = max(res, sum(client))
	}
	return res
}

func max(x, y int) int {
	if x > y {
		return x
	} else {
		return y
	}
}

func maxArr(arr []int) int {
	res := arr[0]
	for _, el := range arr {
		res = max(res, el)
	}
	return res
}

func sum(arr []int) int {
	res := 0
	for _, el := range arr {
		res += el
	}
	return res
}
